### PR TITLE
Add basic deal stats api server for spacerace slingshot

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

File: cmd/lotus-shed/dealtracker.go [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+)
+
+type dealStatsServer struct {
+	api api.FullNode
+}
+
+var filteredClients map[address.Address]bool
+
+func init() {
+	filteredClients = make(map[address.Address]bool)
```
Comment (pooja) [inline cmd/lotus-shed/dealtracker.go:23]: Is there a way we can filter out miners as well? E.g. there are terabytes of data from Travis' miners that were stored at genesis on Space Race net, but this isn't legit usage data.

File: cmd/lotus-shed/dealtracker.go [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+	for _, a := range []string{"t0112", "t0113", "t0114", "t010089"} {
```
Comment (pooja) [inline cmd/lotus-shed/dealtracker.go:24]: And is there some way that we can make this configurable? E.g. if we want to add more addresses to filter out of the search down the road?

**Review by pooja — APPROVED** [review]
OMG @whyrusleeping you have totally made my day. Thank you for this one!!!!

File: cmd/lotus-shed/dealtracker.go (hunk 1 above)
Comment (whyrusleeping) [reply cmd/lotus-shed/dealtracker.go:23]: So, i'm only looking at deals on chain. Do the genesis miners have deals for their data?

File: cmd/lotus-shed/dealtracker.go (hunk 2 above)
Comment (whyrusleeping) [reply cmd/lotus-shed/dealtracker.go:24]: yeah, I can make it into an environment variable

File: cmd/lotus-shed/dealtracker.go (hunk 1 above)
Comment (pooja) [reply cmd/lotus-shed/dealtracker.go:23]: Yeah I believe they do have deals on chain!

File: cmd/lotus-shed/dealtracker.go (hunk 2 above)
Comment (pooja) [reply cmd/lotus-shed/dealtracker.go:24]: Awesome!

**Comment by ribasushi** [comment]
🤯
I did not realize the dealdata is available from the state at all, sigh
Serves me right for not asking.

This will totally work, and obviates need for chainwatch entirely. Thank you @whyrusleeping, you totally saved the day!

**Review by magik6k — APPROVED** [review]
(pushed the env var thing)

**Pull request merged**